starboard/nplb/compiler_compliance/cpp_compiler_version_check.cc

```diff
@@ -14,15 +14,20 @@
 
 #include <iostream>
 
+namespace {
+
+std::string error_message = "We compile above Starboard targets like Cobalt, Nplb  with "
```

Comment:
This is not working as `static_assert` seems to be compiler time checking. The second argument cannot be a variable. Otherwise following error is shown:
```
| ../../starboard/nplb/compiler_compliance/cpp_compiler_version_check.cc:31:22: error: expected string-literal before 'error_message'
|    31 | static_assert(false, error_message);
```

It works with following change
```
#define error_message "We compile above Starboard targets like Cobalt, Nplb with " \
    "clang17 or higher. If you are building with a non-Cobalt " \
    "provided toolchain you may run into this error. To fix this error" \
    "build libnplb using the Evergreen toolchain : " \
    "cobalt.dev/development/setup-raspi"
```

Reply:
Thanks!

I verified this time that i get an error message if i flip the error conditional:
../../starboard/nplb/compiler_compliance/cpp_compiler_version_check.cc:27:15: error: static assertion failed due to requirement '17 < 17': We compile above Starboard targets like Cobalt, Nplb with clang17 or higher. If you are building with a non-Cobalt provided toolchain you may run into this error. To fix this errorbuild libnplb using the Evergreen toolchain : cobalt.dev/development/setup-raspi
static_assert(__clang_major__ < 17, COBALT_CLANG_ERROR_MSG);